### PR TITLE
Add TimedRotatingFileHandler

### PR DESCRIPTION
**Pull request by aminalaee**
https://github.com/python/cpython/blob/main/Lib/logging/handlers.py#L202

There are minor edge cases missing tests but I'll add them in a separate PR to make this process faster.

And also any reason why you are keeping the commit history? For example this PR will create 3 commits which we can squash to one.

### CODE REVIEW

File: tests/unit/test_filehandler.py [hunk 1]
```diff
@@ -155,7 +154,144 @@ def test_baserotatinghandler_callable_namer(tmp_path):
...
+    handler = TimedRotatingFileHandler(log_file, when="S", backupCount=1)
+    logger = picologging.getLogger("test")
+    logger.setLevel(picologging.DEBUG)
+    logger.addHandler(handler)
+
+    logger.warning("test")
+    handler.rollover_at = time.time() - 1
+    logger.warning("test")
+    
+    assert len(os.listdir(tmp_path)) == 2
```
Comment (aminalaee) [inline tests/unit/test_filehandler.py:195]: The rollover should remove the existing files if they match the log format to keep backupCount.

**Comment by tonybaloney** [comment]
WRT your question -- I don't like squashing commits, just a personal preference. I can see some value in having cleaner history but it makes merging harder and blames redundant when trying to understand the reason for that particular change

**Review by tonybaloney — APPROVED** [review]

**Comment by tonybaloney** [comment]
Great stuff, thanks
